opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/AlarmPersisterImpl.java

```diff
@@ -122,15 +125,24 @@ public OnmsAlarm persist(Event event) {
         return alarm;
     }
 
-    private OnmsAlarm addOrReduceEventAsAlarm(Event event) {
-        final OnmsEvent e = m_eventDao.get(event.getDbid());
-        if (e == null) {
+    private OnmsAlarm addOrReduceEventAsAlarm(Event event) throws IllegalStateException {
+        
+        final OnmsEvent persistedEvent = m_eventDao.get(event.getDbid());
+        if (persistedEvent == null) {
             throw new IllegalStateException("Event with id " + event.getDbid() + " was deleted before we could retrieve it and create an alarm.");
         }
 
         final String reductionKey = event.getAlarmData().getReductionKey();
         LOG.debug("addOrReduceEventAsAlarm: looking for existing reduction key: {}", reductionKey);
-        OnmsAlarm alarm = m_alarmDao.findByReductionKey(reductionKey);
+        
+        String key = reductionKey;
+        String clearKey = event.getAlarmData().getClearKey();
+        
+        if (!m_legacyAlarmState && clearKey != null && event.getAlarmData().getAlarmType() == 2) {
```

Comment:
We should use the `OnmsAlarm.RESOLUTION_TYPE` constant, or the `isResolutionEvent` utility method you've added here.

Reply:
√